_posts/en/newsletters/2024-08-09-newsletter.md

```diff
@@ -0,0 +1,342 @@
+    significantly.
+
+  - *Requires pool miners use a private template from the pool:* not
+    only would a template be required to prevent the 100% invalid shares
+    attack but the pool would need to keep that template secret from
+    pool miners until after all shares generated using that template
+    were received.  The pool could use that to trick miners into
+    producing PoW for transactions the miners would object to.  However,
+    expired templates could be published to allow auditing.  Most modern
+    pools generate a new template every few seconds, so any auditing
+    could be performed in near real-time, preventing a malicious pool
+    from tricking its miners for more than a few seconds.
+
+  - *Requires share submission:* one of the advantages of Stratum v2 (in
+    certain operating modes) is that an honest pool miner who finds a
+    block for the pool can immediately broadcast it to the Bitcoin P2P
+    network, allowing it to begin propagating even before the
+    corresponding share has reached the pool server.  With oblivious
+    shares, the share would need to be received by the pool and
+    converted into a full block before it could be broadcast.
+
+  Towns concludes by describing two of the motivations for fixing the block
+  withholding attack: it affects small pools more than large pools, and
+  it costs almost nothing to attack pools that allow anonymous miners,
+  whereas pools that require miners to identify themselves can ban known
+  attackers.  Fixing block withholding could help Bitcoin mining to
+  become more decentralized.
+
+- **Statistics on compact block reconstruction:** developer 0xB10C
+  [posted][0xb10c compact] to Delving Bitcoin about the recent
+  reliability of [compact block][topic compact block relay]
+  reconstruction.  Many relaying full nodes have been using [BIP152][]
+  compact block relay since the feature was added to Bitcoin Core 0.13.0
+  in 2016.  This allows two peers which have already shared some
+  unconfirmed transactions to use a short reference to those
+  transactions when they are confirmed in a new block rather than
+  re-transmitting the entire transaction.  This significantly reduces
+  bandwidth and, by doing so, also reduces latency---allowing new blocks
+  to propagate more quickly.
+
+  Faster propagation of new blocks decreases the number of accidental
+  blockchain forks.  Fewer forks reduces the amount of proof-of-work that
+  is wasted and reduces the number of _block races_ that benefit larger
+  mining pools over smaller pools, helping to make Bitcoin more secure
+  and more decentralized.
+
+  However, sometimes new blocks include transactions that a node has not
+  seen before.  In that case, the node receiving a compact block usually
+  needs to request those transactions from the sending peer and then
+  wait for the peer to respond.  This slows down block propagation.
+  Until a node has all of the transactions in a block, that block can't
```

Comment:
```suggestion
  Until a node has all the transactions in a block, that block can't
```

Reply:
Leaving as-is.  It sounds more correct to me with the preposition.